pyproject.toml

```diff
@@ -0,0 +1,50 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools.packages.find]
+exclude = ["htmlcov"] # Exclude the coverage report file from setuptools package finder
+
+[project]
+name = "WSIMOD"
+description = "WSIMOD is for simulating water quality and quantity"
+readme = "README.md"
+version = "0.2.0"
+license = {file = "LICENSE"}
+authors = [
+    { name = "Imperial College London RSE Team", email = "ict-rse-team@imperial.ac.uk" },
+    { name = "Barnaby Dobson", email = "b.dobson@imperial.ac.uk" }
+]
+requires-python = ">=3.9"
+dependencies = [
+    "PyYAML",
+    "tqdm",
+    "dill"
+]
+
+[project.optional-dependencies]
+dev = [
+    "pytest"
+]
+
+demos = [
+    "pandas",
+    "geopandas",
+    "matplotlib",
+    "shapely"
+]
+
+doc = [
```

Comment:
It seems that when requiring to install the documentation requirements you should install both the `demo` and `doc` requirements. See https://github.com/ImperialCollegeLondon/wsi/blob/741500af25fa2ece9f33268b432dc3a0757c57f3/setup.py#L11

I'm not totally sure how to make this work here for a tool that is not in `pypi` (yet). Maybe try:

```toml
doc = [
    "mkdocs",
    ...
    "wsimod[demos]"
]
```

This is OK for `pypi` hosted tools, but I'm not totally sure this will work for `wsimod` yet. Maybe `".[demos]"` would do the trick?

If this is changed, the `requirements-doc.txt` file will need to be recreated.